AWSCore/Networking/AWSURLSessionManager.m

```diff
@@ -186,6 +195,13 @@ - (void)taskWithDelegate:(AWSURLSessionManagerDelegate *)delegate {
 
             [self printHTTPHeadersAndBodyForRequest:delegate.request.task.originalRequest];
 
+            if (!self.session || !self.isSessionValid) {
```

Comment:
Should we move this above line 193 ie before adding the delegate to `self.sessionManagerDelegates`

Reply:
Yes, good call, will do on next commit.